integration-tests/common/test_common.go

```diff
@@ -3,135 +3,237 @@ package common
+		require.NoError(m.TestConfig.T, err, "Failed to set mock adapter value")
 	}
-	var err error
-	testState.Cc.NKeys, testState.Cc.ChainlinkNodes, err = testState.Common.CreateKeys(testState.Common.Env)
-	require.NoError(testState.T, err, "Creating chains and keys should not fail") // TODO; fails here
-	baseURL := testState.Common.L2RPCUrl
-	if !testState.Common.Testnet { // devnet!
-		// chainlink starknet client needs the RPC API url which is at /rpc on devnet
-		baseURL += "/rpc"
+
+	m.TestConfig.Resty = resty.New().SetBaseURL(m.Common.RPCDetails.RPCL2External)
+	m.SetupClients()
+	if m.Common.TestEnvDetails.IsK8s {
+		m.Clients.ChainlinkClient.ChainlinkNodes = m.GetChainlinkNodes()
+		m.Clients.ChainlinkClient.NKeys, m.TestConfig.err = m.Common.CreateNodeKeysBundle(m.Clients.ChainlinkClient.ChainlinkNodes)
+		require.NoError(m.TestConfig.T, m.TestConfig.err)
+	} else {
+		m.Clients.ChainlinkClient.NKeys, m.TestConfig.err = m.Common.CreateNodeKeysBundle(m.Clients.DockerEnv.ClCluster.NodeAPIs())
+		require.NoError(m.TestConfig.T, m.TestConfig.err)
 	}
-	testState.Starknet, err = starknet.NewClient(testState.Common.ChainId, baseURL, lggr, &rpcRequestTimeout)
-	require.NoError(testState.T, err, "Creating starknet client should not fail")
-	testState.OCR2Client, err = ocr2.NewClient(testState.Starknet, lggr)
-	require.NoError(testState.T, err, "Creating ocr2 client should not fail")
-	if !testState.Common.Testnet {
+	lggr := logger.Nop()
+	m.Clients.StarknetClient, m.TestConfig.err = starknet.NewClient(m.Common.ChainDetails.ChainId, m.Common.RPCDetails.RPCL2External, lggr, &rpcRequestTimeout)
+	require.NoError(m.TestConfig.T, m.TestConfig.err, "Creating starknet client should not fail")
+	m.Clients.OCR2Client, m.TestConfig.err = ocr2.NewClient(m.Clients.StarknetClient, lggr)
+	require.NoError(m.TestConfig.T, m.TestConfig.err, "Creating ocr2 client should not fail")
+
+	// If we are using devnet fetch the default keys
+	if !m.Common.TestEnvDetails.Testnet {
 		// fetch predeployed account 0 to use as funder
-		devnet := starknetdevnet.NewDevNet(testState.Common.L2RPCUrl)
-		accounts, err := devnet.Accounts()
-		require.NoError(testState.T, err)
+		m.Clients.DevnetClient = starknetdevnet.NewDevNet(m.Common.RPCDetails.RPCL2External)
+		accounts, err := m.Clients.DevnetClient.Accounts()
+		require.NoError(m.TestConfig.T, err)
 		account := accounts[0]
-
-		err = os.Setenv("PRIVATE_KEY", account.PrivateKey)
-		require.NoError(testState.T, err, "Setting private key should not fail")
-		err = os.Setenv("ACCOUNT", account.Address)
-		require.NoError(testState.T, err, "Setting account address should not fail")
-		testState.Devnet.AutoDumpState() // Auto dumping devnet state to avoid losing contracts on crash
+		m.Account.Account = account.Address
+		m.Account.PrivateKey = account.PrivateKey
 	}
 }
 
 // DeployEnv Deploys the environment
-func (testState *Test) DeployEnv() {
-	testState.Common.SetLocalEnvironment(testState.T)
```

Comment:
If this replaces the scripts I built then let's delete them from `scripts/` and update the integration-tests readme so it's clear what the dependencies are and how to run this